WordPress/Classes/Models/Media.m

```diff
@@ -261,4 +261,20 @@ - (BOOL)hasRemote {
     return self.mediaID.intValue != 0;
 }
 
+- (void)setError:(NSError *)value
+{
+    NSError *error;
+
+    if (value != nil) {
+        // Cherry pick keys that support secure coding. NSErrors thrown from the OS can
+        // contain types that don't adopt NSSecureCoding, leading to a Core Data exception and crash.
+        NSDictionary *userInfo = @{NSLocalizedDescriptionKey: value.localizedDescription};
```

Comment:
Is the `localizedDescription` the only field we'd want to cherry-pick?

Is the code that was assuming everything was conforming to the `NSSecureCoding` protocol something we built? Should that just be modified to something conditional?

e.g.,
```
if ([object respondsToSelector:@selector(supportsSecureCoding)] && [object supportsSecureCoding]) {
```
or similar.

Reply:
> Is the localizedDescription the only field we'd want to cherry-pick?

This is my understanding based on how I've observed we use the errors. Typically we're only caring about the error domain, type, and the description (e.g. to display to the user in a bottom sheet when they need to retry an upload). I had added in the Regression Notes, though, that there's a chance we used another field somewhere.

> Is the code that was assuming everything was conforming to the NSSecureCoding protocol something we built? Should that just be modified to something conditional?

I'd considered checking protocol conformance but I believe it turns into a case where we need to traverse the object graph. For instance, `NSError` conforms to `NSSecureCoding` [[source](https://developer.apple.com/documentation/foundation/nserror?language=objc)], but its `UserInfo` dictionary could contain an arbitrary value that doesn't.

When looking at the overall scope of how we're using these errors, I went with the cherrypicking approach.

Reply:
Yeah, I suppose I assumed that we'd already be traversing the object graph somewhere, but it's probably not our code that would be doing that directly when we are serializing the `NSDictionary`.

The cherry-picking approach seems fine.